src/resources/views/crud/fields/relationship.blade.php

```diff
@@ -0,0 +1,673 @@
+    // and format it to JSON, so that select2 can parse it
+    $current_value = old(square_brackets_to_dots($field['name'])) ?? old($field['name']) ?? $field['value'] ?? $field['default'] ?? '';
+
+    if ($current_value != false) {
+        switch (gettype($current_value)) {
+            case 'array':
+                $current_value = $connected_entity
+                                    ->whereIn($connected_entity_key_name, $current_value)
+                                    ->pluck($field['attribute'], $connected_entity_key_name);
+                break;
+            case 'object':
+                if(! $current_value->isEmpty())  {
+                    $current_value = $current_value
+                                    ->pluck($field['attribute'], $connected_entity_key_name)
+                                    ->toArray();
+                }
+                break;
+            default:
+                $current_value = $connected_entity
+                                ->where($connected_entity_key_name, $current_value)
+                                ->pluck($field['attribute'], $connected_entity_key_name);
+                break;
+        }
+    }
+    $field['value'] = json_encode($current_value);
+
+    //make sure we provide select options in case field is not an ajax field
+    //or setup the url for fetching those options in ajax
+    $field['options'] = $field['options'] ?? [];
+
+    if($field['ajax'] != true) {
+        if (empty($field['options'])) {
+            $field['options'] = $connected_entity::all()->pluck($field['attribute'],$connected_entity_key_name);
+        } else {
+            $field['options'] = call_user_func($field['options'], $field['model']::query()->pluck($field['attribute'],$connected_entity_key_name));
+        }
+    }else{
+        //case field is ajax we check if FetchOperation is setup
+        //and provide the data_source accordingly
+        if(method_exists('fetch'.ucfirst($field['entity']), $crud->model)) {
+            $field['data_source'] = $field['data_source'] ?? url($crud->route.'/fetch/'.$routeEntity);
+        }
+
+        $field['minimum_input_length'] = $field['minimum_input_length'] ?? 2;
+    }
+
+// InlineCreateOperation set this setting to true, we then
+// check if user configured inline create for this field.
+if($crud->has($crud->getOperation().'.inline_create')) {
+
+$activeInlineCreate = !empty($field['inline_create']) ? true : false;
+
+if($activeInlineCreate) {
+    //if user don't specify a different entity in inline_create we assume it's the same from $field['entity']
+    $field['inline_create']['entity'] = $field['inline_create']['entity'] ?? $field['entity'];
+
+    //we check if this field is not beeing requested in some InlineCreate operation.
+    //this variable is setup by InlineCreate modal when loading the fields.
+if(!isset($inlineCreate)) {
+
+    //route to create a new entity
+    $createRoute = route($field['inline_create']['entity']."-inline-create");
```

Comment:
Hmmm we might have a problem with the routes here. I've added ```relationship``` fields inside ArticleCrudController, for both ```category``` and ```tags```. For Category it worked great, because the route name for the ```CategoryCrudController``` is ```category```. So the ```category-inline-create``` route exists.

But for ```tags```, the route is ```tag-inline-create```, but this tries to load ```tags-inline-create```. So it fails:
![Screenshot 2020-02-19 at 14 48 51](https://user-images.githubusercontent.com/1032474/74835893-f7d4c300-5326-11ea-8472-19455b08747d.png)

See the difference between singular and plural here, when I do ```php artisan route:list```:
![Screenshot 2020-02-19 at 14 49 25](https://user-images.githubusercontent.com/1032474/74836015-32d6f680-5327-11ea-84a8-7d1f6a493c6f.png)

![Screenshot 2020-02-19 at 14 49 49](https://user-images.githubusercontent.com/1032474/74836025-35d1e700-5327-11ea-8e02-04e4b59b87db.png)

The **route** is always singular.
But the **relationship name** can also be plural (for n-n relationships).

Hence, the bug when trying to CreateInline for n-n relationships.

Reply:
Hey @tabacitu,  In field definition setup: `inline_create => ['entity' => 'tag']`

Reply:
True! Works like a freakin' charm 🎉 ! Please leave this open though, so that I can change the docs accordingly 😄